com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/MSISDNBlacklistWhitelistMediator.java

```diff
@@ -0,0 +1,115 @@
+package com.wso2telco.dep.common.mediation;
+
+import com.wso2telco.dep.common.mediation.constant.MSISDNConstants;
+import com.wso2telco.dep.common.mediation.service.APIService;
+import com.wso2telco.dep.common.mediation.util.ContextPropertyName;
+import com.wso2telco.dep.common.mediation.util.ErrorConstants;
+import com.wso2telco.dep.common.mediation.util.ExceptionType;
+import org.apache.http.HttpStatus;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.mediators.AbstractMediator;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MSISDNBlacklistWhitelistMediator extends AbstractMediator {
+
+    private void setErrorInContext(MessageContext synContext, String messageId,
+                                   String errorText, String errorVariable, String httpStatusCode,
+                                   String exceptionType) {
+
+        synContext.setProperty(ContextPropertyName.MESSAGE_ID, messageId);
+        synContext.setProperty("mediationErrorText", errorText);    // mediationErrorText  ContextPropertyName.ERROR_TEXT
+        synContext.setProperty(ContextPropertyName.ERROR_VARIABLE, errorVariable);
+        synContext.setProperty(ContextPropertyName.HTTP_STATUS_CODE, httpStatusCode);
+        synContext.setProperty(ContextPropertyName.EXCEPTION_TYPE, exceptionType);
+    }
+
+    public boolean mediate(MessageContext messageContext) {
+
+        String msisdn = (String) messageContext.getProperty("paramValue");
+        String paramArray = (String) messageContext.getProperty("paramArray");
```

Comment:
Can move this initialization to around to line number 99 since the variable is only used if an exception has occurred.

Reply:
Done